### PR TITLE
feat: push and pull subcommand for model command

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR fixes #
- [x] Implement command to Push models to OCI Registry
- [x]  Implement command to Pull models from OCI Registry
- [ ]  Develop test cases to ensure command functionality

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@RipulHandoo will you update the slide that covers this behavior in the Meshery Architecture deck?

File: mesheryctl/internal/cli/root/system/model.go [hunk 1]
```diff
@@ -413,6 +426,120 @@ mesheryctl exp model search [query-text]
...
+mesheryctl exp model push [path-to-model] --username [username] --password [password] --registry [registry] --tag [tag] --repository [repository]
+	`,
+	// skip preRunE as it is not required for this command
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
+		}
+
+		// path to model
+		modelPath := args[0]
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/system/model.go:445]: Please use `Args` attribute to control the validation of args, and return a meshkit error, if it fails validation.

File: mesheryctl/internal/cli/root/system/model.go [hunk 2]
```diff
@@ -413,6 +426,120 @@ mesheryctl exp model search [query-text]
...
+
+		// Push logic
+		fs, err := file.New(".")
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/system/model.go:448]: What's the location of the created file? The current working directory?
`use os.TempDir() or ~/.meshery`

File: mesheryctl/internal/cli/root/system/model.go [hunk 3]
```diff
@@ -413,6 +426,120 @@ mesheryctl exp model search [query-text]
...
+		if err != nil {
+			return err
+		}
+		ctx := context.Background()
+
+		mediaType := "application/vnd.test.folder"
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/system/model.go:454]: Is this accurate?

File: mesheryctl/internal/cli/root/system/model.go [hunk 4]
```diff
@@ -413,6 +426,120 @@ mesheryctl exp model search [query-text]
...
+		fileNames := []string{modelPath}
+		fileDescriptors := make([]v1.Descriptor, 0)
+		for _, name := range fileNames {
+			fileDescriptor, err := fs.Add(ctx, name, mediaType, "")
+			if err != nil {
+				panic(err)
+			}
+			fileDescriptors = append(fileDescriptors, fileDescriptor)
+			fmt.Printf("file descriptor for %s: %v\n", name, fileDescriptor)
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/system/model.go:463]: meshkit logger.

**Review by MUzairS15 — COMMENTED** [review]
Move the pull and push logic to MeshKit as they will be used heavily across components.

Can you please add tests for pushing and pulling to public OCI registries?

**Comment by leecalcote** [comment]
merge conflict

**Comment by leecalcote** [comment]
@RipulHandoo merge conflict

File: mesheryctl/internal/cli/root/system/model.go (hunk 4 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/model.go:463]: @RipulHandoo feedback from two weeks ago is ready for incorporation.

**Comment by leecalcote** [comment]
A few lint checks are standing in the way, it seems.

**Comment by RipulHandoo** [comment]
> Move the pull and push logic to MeshKit as they will be used heavily across components.
> 
> Can you please add tests for pushing and pulling to public OCI registries?

Sure, I can add tests for pushing and pulling models to public OCI registries. However, to properly authenticate and test these functionalities, we'll need access to a username and password.

**Pull request closed without merging**